src/main/java/org/elasticsearch/percolator/SingleDocumentPercolatorIndex.java

```diff
@@ -56,7 +57,8 @@ public void prepare(PercolateContext context, ParsedDocument parsedDocument) {
             try {
                 // TODO: instead of passing null here, we can have a CTL<Map<String,TokenStream>> and pass previous,
                 // like the indexer does
-                TokenStream tokenStream = field.tokenStream(parsedDocument.analyzer(), null);
+                Analyzer analyzer = context.mapperService().documentMapper(parsedDocument.type()).mappers().indexAnalyzer();
```

Comment:
I think the context of TODO item above is kind of lost here. Maybe move the analyzer assignment up so it would be before TODO?